src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -137,6 +137,34 @@ UCC_CLASS_CLEANUP_FUNC(ucc_tl_ucp_context_t)
 
 UCC_CLASS_DEFINE(ucc_tl_ucp_context_t, ucc_tl_context_t);
 
+ucc_status_t ucc_tl_ucp_mem_map(void *addr, size_t length, ucs_memory_type_t mem_type,
```

Comment:
maybe call the nd ucc_tl_ucp_populate_rcache ? or smth like that. Otherwise, it really looks weird that "mem_map" fn does map+unmap. If rache is not available (for some reason), then the effect is that memory is de-registered.

Reply:
changed the func name.
yes, this is not optimization when rcache is not there.